src/main/java/com/duke/task/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package com.duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Deadline Task.
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    /**
+     * Creates a Deadline Task with the given description and deadline time.
+     *
+     * @param description The description of the deadline event.
+     * @param by          The time by which the deadline event terminates.
+     * @throws DateTimeParseException When the date input is invalid.
+     */
+    public Deadline(String description, String by) throws DateTimeParseException {
```

Comment:
The parameter name "by" doesn't give an intuitive description of what the String stands for. Perhaps could rename to "timing" or "deadline"